config/config_mgmt.py

```diff
@@ -430,65 +478,84 @@ def mergeItems(it1, it2):
         return D1
 
     """
-    Create a defConfig for given Ports from Default Config File.
+    search Relevant Keys in Config using DFS, This function is mainly
+    used to search port related config in Default ConfigDbJson file.
+    In: Config to be searched
+    skeys: Keys to be searched in In Config i.e. search Keys.
+    Out: Contains the search result
     """
-    def getDefaultConfig(self, ports=list()):
+    def searchKeysInConfig(self, In, Out, skeys):
+
+        found = False
+        if isinstance(In, dict):
+            for key in In.keys():
+                #print("key:" + key)
+                for skey in skeys:
+                    # pattern is very specific to current primary keys in
+                    # config DB, may need to be updated later.
+                    pattern = '^' + skey + '\|' + '|' + skey + '$' + \
+                        '|' + '^' + skey + '$'
+                    #print(pattern)
+                    reg = re.compile(pattern)
+                    #print(reg)
+                    if reg.search(key):
+                        # In primary key, only 1 match can be found, so return
+                        # print("Added key:" + key)
+                        Out[key] = In[key]
+                        found = True
+                        break
+                # Put the key in Out by default, if not added already.
+                # Remove later, if subelements does not contain any port.
+                if Out.get(key) is None:
+                    Out[key] = type(In[key])()
+                    if self.searchKeysInConfig(In[key], Out[key], skeys) == False:
```

Comment:
I don't see searchKeysInCOnfig method returning anything. Is it supposed to return "found"?

Reply:
> I don't see searchKeysInConfig method returning anything. Is it supposed to return "found"?

Sorry, I see it below that we are returning "found"